### PR TITLE
PEM private key pattern added

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Use ``MII[A-Za-f] `` pattern (0x30 0x82 0x0_ - ASN1 sequence) to find private key in single line and in xml multiline
- added algorithmic filter to load the pattern without password

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -336,7 +339,8 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v3
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
+          ref: mii
```
Comment (babenek) [inline .github/workflows/benchmark.yml:343]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -11,320 +11,323 @@ on:
...
-          if [ $LOW_DELTA -ge ${d} ]; then
-            echo "Almost the same."
-          elif [ $THRESHOLD -lt ${d} ]; then
-            echo "Significantly Slowdown."
-            exit_code=1
+      - name: Checkout CredData
+        uses: actions/checkout@v4
+        with:
+          repository: babenek/CredData
+          ref: mii
```
Comment (babenek) [inline .github/workflows/benchmark.yml:169]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -11,320 +11,323 @@ on:
...
-    - name: Verify benchmark scores on push event
-      if: ${{ 'pull_request' != github.event_name }}
-      # update cicd/benchmark.txt with uploaded artifact if a difference is found
-      run: |
-        diff --ignore-all-space --ignore-blank-lines CredSweeper/cicd/benchmark.txt benchmark.txt
+      - name: Checkout CredData
+        uses: actions/checkout@v4
+        with:
+          repository: babenek/CredData
+          ref: mii
```
Comment (babenek) [inline .github/workflows/benchmark.yml:67]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 4]
```diff
@@ -11,320 +11,323 @@ on:
...
 
-    - name: Checkout CredData
-      uses: actions/checkout@v4
-      with:
-        repository: Samsung/CredData
+      - name: Checkout CredData
+        uses: actions/checkout@v4
+        with:
+          repository: babenek/CredData
+          ref: mii
```
Comment (babenek) [inline .github/workflows/benchmark.yml:26]:
```suggestion
          repository: Samsung/CredData
```

**Review by babenek — COMMENTED** [review]
TODO: rollback after https://github.com/Samsung/CredData/pull/78

**Review by Yullia — APPROVED** [review]

**Review by Yullia — APPROVED** [review]
Approved

File: tests/filters/test_value_base64_key_check.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                "/O 2 h d p 6 l Z vSWVt/KD+HJMRIDVqj39DphuU8f0cqJkpLoP5xcetOH"
+                "       /XVnIVQXm5AkEA4gc5HjeggbJkj4bJg/ROmKlLgMKalg0LS7Z3"
+                "\tYXIhIGJcwE27ERB5DjqMJp38Yvz8WqUchNFqF4fxSEYTWVIiMwJBAN"
+                "\t\t\t+xiNpJtX3qWI4T5iOQM7pE8Ngx+olpOpZkwQ6jBzNyTfaZZT\""
+                "XpmvzvBZKWgkvxAE3BaeHjuXKlRYp7dVTXi2cCQEeKL7rvsl66nveKmueoAO \\\\"
+                "\nRy2RH1Bkat9EBPjXOLUw7T8zVupOHey+IEB+4FHSgb"
+                "\r\n\tmKf8EI29Vo4CGg1dGKyxwV8CQAqZ/"
+                "\r   WRrDeDz1FKXetbApNL1JXz5kWglxpMOH2A0NckoZ62CQ5u0gJXwNhFmdLJor5z"
+                "/x6bfBhD66DrR8xOcS40CQBoGNvyzjHXUdx64Yxmtz2FXq0fYI8J8ChgNfTe6l6d1nzEQ7i0SspM45L"
+                "/n5tgyrBBGcqzQ/WBL1C8Ny5K+RxY='''\n  -----END")
```
[the remark this answers is not in the document]
Comment (babenek) [reply tests/filters/test_value_base64_key_check.py:39]: It is just for test. Any garbage symbols may break the filter verification.

[the remark this answers is not in the document]
Comment (csh519) [reply tests/filters/test_value_base64_key_check.py:39]: Ah I see.

**Review by csh519 — APPROVED** [review]
Thank you for adding `BASE64 Private Key` rule.

LGTM :+1:

**Pull request merged**